jprod_lithdet update for COBALTv3

feida6996 opened the pull request:
This addresses discussion point https://github.com/NOAA-CEFI-Regional-Ocean-Modeling/ocean_BGC/issues/35 by adjusting the production of lithogenic detritus and conducting tests.

The tests show that the new expression has eliminated the occurrence of abnormally high values of lithogenic detritus and the subsequent flux crash. Instead, the flux stabilizes and reaches equilibrium after approximately 10 years.

We will also verify other key variables to ensure that they exhibit expected behavior after this change.

<img width="458" alt="Picture1" src="https://github.com/NOAA-CEFI-Regional-Ocean-Modeling/ocean_BGC/assets/127246088/60abaf7d-bf20-4d75-a2df-e4b359a6f2d7">

Commit:
jprod_lithdet update for COBALTv3

yichengt900 commented on generic_tracers/generic_COBALT.F90:
```diff
@@ -8070,7 +8070,8 @@ subroutine generic_COBALT_update_from_source(tracer_list,Temp,Salt,rho_dzt,dzt,h
        cobalt%jprod_lithdet(i,j,k)=( cobalt%total_filter_feeding(i,j,k)/ &
-                                   ( phyto(LARGE)%f_n(i,j,k) + phyto(DIAZO)%f_n(i,j,k) + epsln ) * &
+                                   ( phyto(LARGE)%f_n(i,j,k) + phyto(DIAZO)%f_n(i,j,k) + &
+                                     0.8*phyto(MEDIUM)%f_n(i,j,k) + 0.3*phyto(SMALL)%f_n(i,j,k) + epsln) * &
```
@feida6996 @charliestock , Do you think there's a chance that we may need to retune these parameters (0.8, 0.3)? If so, should we consider putting them in the add_param [Remineralization section](https://github.com/NOAA-CEFI-Regional-Ocean-Modeling/ocean_BGC/blob/10613509b0c50400dc86c9b18bc5e92cc1a1a951/generic_tracers/generic_COBALT.F90#L5057-L5080)

yichengt900 reviewed (commented):
@feida6996, thanks for the PR! The new relationship looks good to me. Don't worry about the RT test failure for now. It's expected since your changes result in different modeled answers. I only have one small comment. @charliestock, should we consider conducting a test run in the regional domain (e.g., NWA12) just for a sanity check before merging this PR? Also, @feida6996, feel free to add more checks (e.g., surface chl-a) if you have any.

andrew-c-ross committed:
Fix setting of as_param and allow generic_COBALT to set as_param_COBALT (#42)

* Fix setting of as_param and allow generic_COBALT to set its own as_param if needed

* Update default air-sea param to Wanninkhoff 2014

* Switch workflow to develop branch

yichengt900 commented:
Ok here are some updated preliminary results (1993-1997) from NWA12 with the new expression:

surface alk, dic, and omega:
![alk_dic_omega](https://github.com/NOAA-CEFI-Regional-Ocean-Modeling/ocean_BGC/assets/143743249/4922ce6b-e6a1-436d-b75a-3615f5a7f144)

Surface chl:
![regional_chl](https://github.com/NOAA-CEFI-Regional-Ocean-Modeling/ocean_BGC/assets/143743249/4af6cb93-3c71-4d10-b153-8019c46f271b)

So far, everything looks ok to me.

yichengt900 committed:
Generic cobalt id cleanup and remove legacy natu (#40)

* remove unused ids in generic_COBALT type

* removed legacy ntau=tau and tau=1

yichengt900 commented:
@feida6996, based on the short-term results from the regional domain (see above Figs), I can let your PR sneak in this time if you can make the following changes:

1. Follow this [instruction](https://github.com/NOAA-CEFI-Regional-Ocean-Modeling/ocean_BGC/wiki/How-to-clone,-create-branch,-commit-changes,-push-changes,-and-open-a-pull-request#41-check-your-feature-branch-is-up-to-date-with-the-main-branch-before-merging) to merge the recent updates from main branch to your feature branch;

2. After 1, copy the following and update the [baseline answers](https://github.com/feida6996/ocean_BGC/blob/litdet/.github/ref/OM4.single_column.COBALT.p4/docker-linux-gnu/ocean.stats) in your feature branch:
```
  Step,       Day,  Truncs,      Energy/Mass,      Maximum CFL,  Mean Sea Level,  Total Mass,  Mean Salin, Mean Temp, Frac Mass Err,   Salin Err,    Temp Err
            [days]                 [m2 s-2]           [Nondim]       [m]             [kg]         [PSU]      [degC]       [Nondim]        [PSU]        [degC]
     0,  731595.000,     0, En 4.0531850015098383E-27, CFL  0.00000, SL -0.0000E+00, M 6.97073E+15, S 35.2427, T  6.2462, Me  0.00E+00, Se  0.00E+00, Te  0.00E+00
     6,  731595.250,     0, En 2.0223578954766472E-06, CFL  0.00281, SL -3.4291E-06, M 6.97073E+15, S 35.2427, T  6.2460, Me -8.57E-10, Se -2.85E-07, Te -2.66E-06
    12,  731595.500,     0, En 4.2897044161823591E-06, CFL  0.00294, SL -6.4127E-06, M 6.97073E+15, S 35.2427, T  6.2457, Me -7.46E-10, Se -6.81E-07, Te -6.81E-06
    18,  731595.750,     0, En 1.2185516288182668E-05, CFL  0.00794, SL -9.2141E-06, M 6.97073E+15, S 35.2427, T  6.2459, Me -7.00E-10, Se -9.89E-07, Te -1.00E-05
    24,  731596.000,     0, En 2.4391883629687393E-05, CFL  0.00870, SL -1.3774E-05, M 6.97073E+15, S 35.2427, T  6.2457, Me -1.14E-09, Se -1.40E-06, Te -1.42E-05
    30,  731596.250,     0, En 2.9283640524189851E-05, CFL  0.00724, SL -1.8458E-05, M 6.97073E+15, S 35.2427, T  6.2454, Me -1.17E-09, Se -1.93E-06, Te -1.96E-05
    36,  731596.500,     0, En 3.6652617806009578E-05, CFL  0.00746, SL -2.3211E-05, M 6.97073E+15, S 35.2427, T  6.2451, Me -1.19E-09, Se -2.37E-06, Te -2.42E-05
    42,  731596.750,     0, En 4.2253748566466643E-05, CFL  0.00812, SL -2.7653E-05, M 6.97073E+15, S 35.2427, T  6.2454, Me -1.11E-09, Se -2.79E-06, Te -2.86E-05
    48,  731597.000,     0, En 4.7781350583834568E-05, CFL  0.00941, SL -3.1143E-05, M 6.97073E+15, S 35.2427, T  6.2453, Me -8.73E-10, Se -3.05E-06, Te -3.14E-05
```

3. Add a commit for unit correction if you have one.

CC @charliestock , please consider adding a comment for this new expression.

Commit:
change unit of lithogenic detrital flux and production integral in upper 100m

charliestock commented:
Hi Yi-Cheng, Fei,

Sorry for the long response time on this.  I was pulled away on a few other things this week.  The solution looks good to me.  While the jprod_lithdet fix will change answers, I expect it to be small.

As we discussed on Tuesday, it is possible that the coefficients will change if the model is reformulated.  The current rationale is as follows:  The filter feeders responsible for converting lithogenic material to lithogenic detritus are either medium or large zooplankton (but not small).  Most of the small phytoplankton are consumed by microzooplankton.  The value of 0.3 thus reflects the relative availability of small phytoplankton to medium zooplankton (i.e., filter feeders) rather than small zooplankton.  This roughly comes out to 0.4/(0.4 + 1.0) = 0.29, which I rounded to 0.3 to not give a sense of false precision.

For medium phytoplankton, the analogous calculation is: (1.0 + 0.4)/(0.4 + 1.0 + 0.4) = 0.78, again rounded to 0.8 to not give a sense of false precision.

I'd also emphasize that the production of lithdet from lith is a coarse scaling that basically ensures that lithogenic material is acting to preserve detritus near regions with high dust and river inputs.  There is room for improvement and sensitivities here, but it is really just a first order transfer function.

Once this pull request is in, I will add a comment to clarify the rationale for these numbers.

feida6996 committed:
Update ocean.stats

feida6996 commented:
> @feida6996, based on the short-term results from the regional domain (see above Figs), I can let your PR sneak in this time if you can make the following changes:
> 
> 1. Follow this [instruction](https://github.com/NOAA-CEFI-Regional-Ocean-Modeling/ocean_BGC/wiki/How-to-clone,-create-branch,-commit-changes,-push-changes,-and-open-a-pull-request#41-check-your-feature-branch-is-up-to-date-with-the-main-branch-before-merging) to merge the recent updates from main branch to your feature branch;
> 2. After 1, copy the following and update the [baseline answers](https://github.com/feida6996/ocean_BGC/blob/litdet/.github/ref/OM4.single_column.COBALT.p4/docker-linux-gnu/ocean.stats) in your feature branch:
> 
> ```
>   Step,       Day,  Truncs,      Energy/Mass,      Maximum CFL,  Mean Sea Level,  Total Mass,  Mean Salin, Mean Temp, Frac Mass Err,   Salin Err,    Temp Err
>             [days]                 [m2 s-2]           [Nondim]       [m]             [kg]         [PSU]      [degC]       [Nondim]        [PSU]        [degC]
>      0,  731595.000,     0, En 4.0531850015098383E-27, CFL  0.00000, SL -0.0000E+00, M 6.97073E+15, S 35.2427, T  6.2462, Me  0.00E+00, Se  0.00E+00, Te  0.00E+00
>      6,  731595.250,     0, En 2.0223578954766472E-06, CFL  0.00281, SL -3.4291E-06, M 6.97073E+15, S 35.2427, T  6.2460, Me -8.57E-10, Se -2.85E-07, Te -2.66E-06
>     12,  731595.500,     0, En 4.2897044161823591E-06, CFL  0.00294, SL -6.4127E-06, M 6.97073E+15, S 35.2427, T  6.2457, Me -7.46E-10, Se -6.81E-07, Te -6.81E-06
>     18,  731595.750,     0, En 1.2185516288182668E-05, CFL  0.00794, SL -9.2141E-06, M 6.97073E+15, S 35.2427, T  6.2459, Me -7.00E-10, Se -9.89E-07, Te -1.00E-05
>     24,  731596.000,     0, En 2.4391883629687393E-05, CFL  0.00870, SL -1.3774E-05, M 6.97073E+15, S 35.2427, T  6.2457, Me -1.14E-09, Se -1.40E-06, Te -1.42E-05
>     30,  731596.250,     0, En 2.9283640524189851E-05, CFL  0.00724, SL -1.8458E-05, M 6.97073E+15, S 35.2427, T  6.2454, Me -1.17E-09, Se -1.93E-06, Te -1.96E-05
>     36,  731596.500,     0, En 3.6652617806009578E-05, CFL  0.00746, SL -2.3211E-05, M 6.97073E+15, S 35.2427, T  6.2451, Me -1.19E-09, Se -2.37E-06, Te -2.42E-05
>     42,  731596.750,     0, En 4.2253748566466643E-05, CFL  0.00812, SL -2.7653E-05, M 6.97073E+15, S 35.2427, T  6.2454, Me -1.11E-09, Se -2.79E-06, Te -2.86E-05
>     48,  731597.000,     0, En 4.7781350583834568E-05, CFL  0.00941, SL -3.1143E-05, M 6.97073E+15, S 35.2427, T  6.2453, Me -8.73E-10, Se -3.05E-06, Te -3.14E-05
> ```
> 
>  3. Add a commit for unit correction if you have one.
> 
> CC @charliestock , please consider adding a comment for this new expression.

Hi @yichengt900, 

I double checked and the feature branch is up to date with the main branch. The baseline answer was updated. Two unit changes were also added.

yichengt900 commented on .github/ref/OM4.single_column.COBALT.p4/docker-linux-gnu/ocean.stats:
```diff
@@ -1,11 +1,11 @@
-  Step,       Day,  Truncs,      Energy/Mass,      Maximum CFL,  Mean Sea Level,  Total Mass,  Mean Salin, Mean Temp, Frac Mass Err,   Salin Err,    Temp Err
+   Step,       Day,  Truncs,      Energy/Mass,      Maximum CFL,  Mean Sea Level,  Total Mass,  Mean Salin, Mean Temp, Frac Mass Err,   Salin Err,    Temp Err
```
@feida6996, I don't want to be so picky, but the RT test crashed and complained about the additional white space in the first line. I believe it will pass if you remove it

feida6996 committed:
Merge branch 'NOAA-CEFI-Regional-Ocean-Modeling:dev/cefi' into litdet

yichengt900 committed:
fix ocean.stats

@feida6996 Sorry to directly commit to your branch. I removed the extra white space in your ocean.stats.

yichengt900 reviewed (approved):
@feida6996, thanks for opening this PR. The results look promising. I took the liberty of fixing a minor issue to help it pass automated testing, though typically this would be the responsibility of the person opening the PR. I approve.
